### PR TITLE
🐛 Source Amazon Ads: test coverage more than 90% + improve test speed

### LINKED ISSUE #11367: 🐛 Source Amazon Ads: test coverage more than 90% + improve test speed
- Need to improve unit test coverage at least 90%
- need to improve test speed
- need to update pendulum 1.5 -> 2.1
- Improve error handling
- Improve acceptance test organization

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Improve unit test coverage at least 90%
Improve test speed
Update pendulum 1.5 -> 2.1
Improve error handling
Improve acceptance test organization

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py [hunk 1]
```diff
@@ -266,6 +270,7 @@ def test_display_report_stream_init_too_many_requests(mocker, test_config):
 )
 @responses.activate
 def test_display_report_stream_backoff(mocker, test_config, modifiers, expected):
+    mocker.patch("time.sleep")
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py:273]: do we save much time?

**Review by davydov-d — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py (hunk 1 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py:273]: 2-5 seconds

**Review by bazarnov — APPROVED** [review]

**Comment by midavadim** [comment]
This update can be released without Airbyte review.

**Review by midavadim — APPROVED** [review]
Approved

**Pull request merged**